[Classification workflow] AssertionError: empty strata detected: {1, 2, 3, 4}

_Reported by @dpoursanidis via email._

Running the classification workflow with the provided dataset results in an error.

<img width="600" height="409" alt="Image" src="https://github.com/user-attachments/assets/c434f7a7-a8b4-4797-938a-dc385ee2e32b" />

```
QGIS version: 3.40.9-Bratislava
QGIS code revision: 69fa2ab303
Qt version: 5.15.13
Python version: 3.12.11
GDAL version: 3.11.3
GEOS version: 3.13.1-CAPI-1.19.2
PROJ version: Rel. 9.6.2, June 4th, 2025
PDAL version: 2.9.0 (git-version: 42b180)
Algorithm started at: 2025-09-16T12:57:43
Algorithm 'Classification workflow' starting…
Input parameters:
{ 'classifier' : 'from sklearn.ensemble import RandomForestClassifier\nclassifier = RandomForestClassifier(n_estimators=100, oob_score=True)', 'dataset' : 'D:/qgis/bin/qgis-ltr-bin/processing_iStmWe/18812817c3934a98a711897f0ceb03b9/outputClassificationDataset.pkl', 'matchByName' : False, 'nfold' : 10, 'openReport' : True, 'outputClassification' : 'TEMPORARY_OUTPUT', 'outputClassificationAccuracy' : 'TEMPORARY_OUTPUT', 'outputClassifier' : 'TEMPORARY_OUTPUT', 'outputClassifierPerformance' : 'TEMPORARY_OUTPUT', 'outputProbability' : 'TEMPORARY_OUTPUT', 'raster' : 'C:/Users/janzandr/Downloads/Neuer Ordner (3)/SG015_clip.tif' }

Traceback (most recent call last):
File "D:\source\QGISPlugIns\enmap-box\enmapbox\typeguard\__init__.py", line 1033, in wrapper
retval = func(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^
File "D:\source\QGISPlugIns\enmap-box\enmapboxprocessing\algorithm\classificationperformancestratifiedalgorithm.py", line 215, in processAlgorithm
stats = stratifiedAccuracyAssessment(stratum, yReference, yMap, h, N_h, classValues, classNames)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "D:\source\QGISPlugIns\enmap-box\enmapbox\typeguard\__init__.py", line 1033, in wrapper
retval = func(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^
File "D:\source\QGISPlugIns\enmap-box\enmapboxprocessing\algorithm\classificationperformancestratifiedalgorithm.py", line 381, in stratifiedAccuracyAssessment
stats = aa_stratified(stratum, reference, map, h, N_h, classValues)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "D:\source\QGISPlugIns\enmap-box\enmapbox\typeguard\__init__.py", line 1033, in wrapper
retval = func(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^
File "D:\source\QGISPlugIns\enmap-box\enmapboxprocessing\algorithm\classificationperformancestratifiedalgorithm.py", line 406, in aa_stratified
assert set(h) == set(stratum), f'empty strata detected: {set(h) - set(stratum)}'
^^^^^^^^^^^^^^^^^^^^^^
AssertionError: empty strata detected: {1, 2, 3, 4}
```

--- a/enmapboxprocessing/algorithm/rasterizecategorizedvectoralgorithm.py
+++ b/enmapboxprocessing/algorithm/rasterizecategorizedvectoralgorithm.py
@@ -217,7 +217,11 @@ class RasterizeCategorizedVectorAlgorithm(EnMAPProcessingAlgorithm):
         # calculate class ids [1..nCategories]
         vector2 = QgsVectorLayer(filename)
         fieldIndex = vector2.fields().indexOf(classFieldName)
-        idOfValue = {value: id for id, value in enumerate(values, 1)}
+        idOfValue = dict()
+        for id, value in enumerate(values, 1):
+            idOfValue[value] = id
+            if value.isnumeric():
+                idOfValue[value + '.0'] = id
 
         n = vector2.featureCount()
         with edit(vector2):

